app/controllers/questionnaire_controller.rb

```diff
@@ -26,12 +26,43 @@ def create_informed_consent
   def create
     response_content = ResponseContent.create!(content: questionnaire_content)
     @response.update_attributes!(content: response_content.id, completed_at: Time.zone.now)
+    check_stop_subscription unless questionnaire_stop_subscription.blank?
     redirect_to(mentor_overview_index_path) && return unless @protocol_subscription.for_myself?
     redirect_to klaar_path
   end
 
   private
 
+  def check_stop_subscription
+    stop_subscription_hash = questionnaire_stop_subscription
+    content = questionnaire_content
+    should_stop = false
+    stop_subscription_hash.each do |key, value|
+      next unless content.key?(key)
+      expected = Response.stop_subscription_token(key, content[key], @response.id)
+      received = value
+      are_equal = ActiveSupport::SecurityUtils.secure_compare(expected, received)
```

Comment:
De meegegeven stop_questionnaire tokens geven aan "Hee, als het antwoord op vraag A C is, stop dan de protocolsubscriptie". We gebruiken one-way hashing met een salt om te voorkomen dat studenten met de stop_questionnaire tokens kunnen tamperen en dan hun dagboekstudie kunnen stoppen.